show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt

```diff
@@ -0,0 +1,195 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components
+
+import android.app.Application
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.CoordinateFormatter
+import com.arcgismaps.geometry.LatitudeLongitudeFormat
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.geometry.UtmConversionMode
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Basemap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.layers.ArcGISTiledLayer
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.R
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class MapViewModel(application: Application) : AndroidViewModel(application) {
+    // set the MapView mutable stateflow
+    val mapViewState = MutableStateFlow(MapViewState())
+
+    var decimalDegrees = mutableStateOf("")
+    var degreesMinutesSeconds = mutableStateOf("")
+    var utm = mutableStateOf("")
+    var usng = mutableStateOf("")
```

Comment:
I did a deep dive on this as well because we usually use `StateFlow` rather than just `State`. I couldn't find a lot of info on what Android recommends as best practice, aside from [this codelab](https://developer.android.com/codelabs/jetpack-compose-state#11). I think in an Android/Compose-only app you are right to use `State`. However, I still think best practice would be to keep the setter private and to use a delegate rather than setting. So ignoring my comment above, the suggested code would be:
```suggestion
    var decimalDegrees by mutableStateOf("")
        private set
    var degreesMinutesSeconds by mutableStateOf("")
        private set
    var utm by mutableStateOf("")
        private set
    var usng by mutableStateOf("")
        private set
```
To be clear, I think you need to use `by` because if you just set the var using `=` then changing the var will not change the value inside the `MutableState` but will instead change the reference to the state itself. Using `by`, these properties can be treated as normal `var`s.

Using the `private set` ensures that only the viewmodel can change the state, but keeps the property publicly visible to observers

Note this requires two imports and anywhere you call eg. `decimalDegrees.value` would be changed to just `decimalDegrees`

Reply:
This was a really helpful comment. Moving forward we should design the VM in a way that only it can change the state but keep it's property publicly visible. 👍